Replace configuration with RPM FileFlags.

talios opened the pull request:
A while ago we had a need/desire to configure our config files with %noreplace, but this wasn't supported or easily exposed via the current PackageDrone library.

This patch adds the full set of RPMFILE_* file flags defined by the current RPM project, and exposes setting those FileFlags via the `FileInformation` class. The existing `CONFIGURATION` and `setConfiguration` / `isConfiguration` code has been left in, but deprecated for now.

Initially I was going to base this PR on the `master` branch, but it seems that doesn't match the 0.14.1 release that's available in Maven Central, so this PR is based off the `0.14.x-release` branch.

CLA has been signed.

ctron commented:
Thank you for this PR. This really is awesome.

I would like to amend a few things in the same PR, if that is fine with you. I would like to understand if  there is any particular reason why you deprecated the "CONFIGURATION" literal? As the enum is already named "FileFlags" it seems a bit redundant to me to prefix them with "RPMFILE_". Also is the "NONE" literal not really needed as the same can be achieved with an empty `EnumSet<FileFlags>`. But maybe I missed something here.

I any case, thank you very much. I definitely want to get this merged soon.

talios commented:
@ctron Cheers - took awhile to get some time away from normal work to look at this again.

A quick reply as I'm just heading to bed shortly. The only reason I used the `RPMFILE_` prefix is that's the constant used in the RPM sources themselves, so mostly keeping a measure of familiarity/consistency but happy to drop the prefix. If we do that, I could drop the `@Deprecated` and also rename `MISSINGOK`/`NOREPLACE` to also mention `CONFIGURATION_`.

Dropping `NONE` probably seems wise given the use of `EnumSet`, initially I left it in for completeness ( likewise the `SPECFILE`, which based on upstream source sounds like it may be removed at some point.

I'll update the PR in the morning with those changes if they sound good to you.

Ideally I'd love to rename `FileFlags` to `FileFlag` as usually enums/classes are named in the singular ( more so now with an `EnumSet<FileFlag>` but renaming is a change beyond the scope of this having downstream changes.

ctron commented:
Cool. Thank you that you will take care of that. I do appreciate it.

I agree with the FileFlag(s) ... But that would break the current API and I don't think this is worth it. Some things are just a bit odd 😉

talios commented:
Updated the PR with:

* Removal of `@Deprecated`
* Dropped `RPMFILE_` prefix from `FileFlags` entries and also removed the `NONE` and `SPECFILE` values.
* Added static `decode` method to take an `int` to an `EnumSet<FileFlags>`

ctron commented:
Thank you very much for the PR. I will try to get out a release as soon as possible.

ctron commented:
The release is done. As this only affects the RPM library, I only did a release on Maven Central.
